app/modes/library/routes.py

```
@@ -0,0 +1,769 @@
+
+    book_progress = {}
+    book_needs_generation = {}
+
+    for book in my_books:
+        # Check current progress first (trust the database-backed system)
+        progress_data = get_generation_progress(book.id)
+
+        needs_generation = False
+        if progress_data['status'] == 'completed':
+            needs_generation = False
+            logger.info(f"Book {book.id} ({book.title}): Status is completed. Marked as complete.")
```

Ruff (0.15.1)

[warning] Logging statement uses f-string

(G004)